Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -120,43 +118,37 @@ namespace AzToolsFramework::ComponentModeFramework
 
                 for (const auto& entityComponent : entity->GetComponents())
                 {
-                    // Find if the component has a component mode using GetNumOfEventHandlers
                     const auto entityComponentIdPair = AZ::EntityComponentIdPair(entity->GetId(), entityComponent->GetId());
                     AddComponentButton(entityComponentIdPair);
                 }
             }
         }
         else if (!newlyDeselectedEntityIds.empty())
         {
-        /*    AZStd::vector<AZ::EntityComponentIdPair> pairIdsToRemove;
-            for (auto componentDataIt : m_addedComponents)
-            {
-                pairIdsToRemove.push_back(componentDataIt.m_pairId);
-            }
-            for (auto pairId : pairIdsToRemove)
-            {
-                RemoveComponentButton(pairId);
-            }*/
-            //v.erase(std::remove_if(v.begin(), v.end(), is_odd), v.end());
-            for (AZStd::vector<ComponentData>::reverse_iterator it = m_addedComponents.rbegin(); it != m_addedComponents.rend(); ++it)
-            {
-                RemoveComponentButton(it->m_pairId);
-            }
+            ClearSwitcher();
+        }
+    }
+
+    void ComponentModeSwitcher::ClearSwitcher()
+    {
+        for (AZStd::vector<ComponentData>::reverse_iterator it = m_addedComponents.rbegin(); it != m_addedComponents.rend(); ++it)
+        {
+            RemoveComponentButton(it->m_pairId);
         }
     }
 
     void ComponentModeSwitcher::AddComponentButton(const AZ::EntityComponentIdPair pairId)
     {
-        // Check if the component has a component mode.
-        const auto numHandlers = AzToolsFramework::ComponentModeFramework::ComponentModeDelegateRequestBus::GetNumOfEventHandlers(pairId);
-        if (numHandlers < 1)
+        // check if the component has a component mode
+        const auto handlerCount = AzToolsFramework::ComponentModeFramework::ComponentModeDelegateRequestBus::GetNumOfEventHandlers(pairId);
+        if (handlerCount < 1)
         {
             return;
         }
```

Comment:
this looks like a condition where this would be a soft error?

Reply:
The process is that it loops through every component on the entity, it there isn't a handler for the ComponentModeDelegateRequestBus it just means that it's a component without Edit mode and we exit without adding a button.